components/shared/compositing/lib.rs

```diff
@@ -486,6 +489,29 @@ impl CrossProcessCompositorApi {
     }
 }
 
+#[derive(Clone)]
+pub struct PainterSurfmanDetails {
+    pub connection: Connection,
+    pub adapter: Adapter,
+    pub gl_type: GlType,
```

Comment:
I think you don't actually need to send `GlType` here as you can trivially get it via the `Connection` with `Connection::gl_api`. That should simplify things slightly.